src/main/java/net/earthcomputer/clientcommands/command/AliasCommand.java

```diff
@@ -0,0 +1,164 @@
+            for(String key: aliasMap.keySet()) {
+                addClientSideCommand(key);
+                dispatcher.register(literal(key)
+                        .executes(ctx -> executeAliasCommand(key, null))
+                        .then(argument("arguments", greedyString())
+                                .executes(ctx -> executeAliasCommand(key, getString(ctx, "arguments")))));
+            }
+        }
+
+    }
+    private static int executeAliasCommand(String aliasKey, String arguments) throws CommandSyntaxException {
+
+        // TODO: add support for optional greedy text arguments(?)
+        String cmd;
+        if(aliasMap != null && aliasMap.containsKey(aliasKey)) {
+            cmd = aliasMap.get(aliasKey);
+        } else {
+            throw NOT_FOUND_EXCEPTION.create(aliasKey);
+        }
+
+        if(arguments!=null) {
+            cmd = String.format(cmd, arguments.split(" "));
+        }
+        assert MinecraftClient.getInstance().player != null;
+        MinecraftClient.getInstance().player.sendChatMessage(cmd);
+
+        return 0;
+    }
+
+    private static int addAlias(String key, String command) throws CommandSyntaxException {
+        ClientPlayNetworkHandler networkHandler = MinecraftClient.getInstance().getNetworkHandler();
+        assert networkHandler != null;
+        var dispatcher = (CommandDispatcher<ServerCommandSource>) (CommandDispatcher<?>) networkHandler.getCommandDispatcher();
+
+        assert aliasMap != null;
+        if(aliasMap.containsKey(key)) {
+            throw ALREADY_EXISTS_EXCEPTION.create(key);
+        }
+        addClientSideCommand(key);
+
+        dispatcher.register(literal(key)
+                .executes(ctx -> executeAliasCommand(key, null))
+                .then(argument("arguments", greedyString())
+                        .executes(ctx -> executeAliasCommand(key, getString(ctx, "arguments")))));
+        aliasMap.put(key, command);
+
+        saveAliases(new TranslatableText("commands.calias.addAlias.success", key));
+        return 0;
+    }
+    private static int listAliases() {
+        if (aliasMap.isEmpty()) {
+            sendFeedback(new TranslatableText("commands.calias.listAliases.noAliasesRegistered"));
+        } else {
+            sendFeedback("commands.calias.listAliases.success", aliasMap.size());
+
+            for(String key: aliasMap.keySet()) {
+                sendFeedback(Formatting.BOLD + key + Formatting.RESET+ ": "+ aliasMap.get(key).replace("%","%%"));
+            }
+        }
+        return 0;
+    }
+    private static int removeAlias(String key) throws CommandSyntaxException {
+        ClientPlayNetworkHandler networkHandler = MinecraftClient.getInstance().getNetworkHandler();
+        assert networkHandler != null;
+        var dispatcher = (CommandDispatcher<ServerCommandSource>) (CommandDispatcher<?>) networkHandler.getCommandDispatcher();
+
+        if(aliasMap != null && aliasMap.containsKey(key)) {
```

Comment:
In think this `if`-statement should be split in two.

Reply:
I would disagree, but it seems the first half can be removed entirely. I don't see a point in throwing a unique exception here.